Processor/SevenZipProcessor.php

```diff
@@ -33,7 +33,9 @@ public function getCompressionCommand($archivePath, $basePath)
             $params[] = '-mx'.$compression_ratio;
         }
 
-        return sprintf('cd %s && 7z a %s %s', $basePath, implode(' ', $params), $archivePath);
+        $binaryFile = escapeshellarg(isset($this->options['executable']) ? $this->options['executable'] : '7z');
```

Comment:
This can be shortened to `$binaryFile = escapeshellarg($this->options['executable'] ?: '7z');`